gcc/cp/contracts.cc

```diff
@@ -2799,20 +2800,25 @@ maybe_reject_param_in_postcondition (tree decl, location_t location)
 }
 
 /* Check if parameters used in postconditions are const qualified on
-   a redeclaration that does not specify contracts.  */
+   a redeclaration that does not specify contracts or on an instantiation
+   of a function template.  */
 
 void
 check_param_in_redecl (tree olddecl, tree newdecl)
```

Comment:
So for the name. I'd like to have this function's name give a hint that it's a contract-related check, because it is. Here's what I suggest:
1) rename maybe_reject_param_in_postcondition to check_param_in_postcondition
2) rename check_param_in_redecl to check_postcondition_in_redecl

We could also entertain something like check_params_in_redecl_postcondition for the latter, but I don't think we need to be that specific in the name what exactly we are checking. The current name is ever so slightly off in another way, too: it checks params, not just one param. :)

Reply:
Also, if you want to make it check_postconditions_in_redecl, that's 100% fine, since there may be more than one. :)